test/metabase/api/card_test.clj

```diff
@@ -2450,3 +2450,21 @@
           (is (not (contains? (task.persist-refresh/job-info-for-individual-refresh)
                               (u/the-id parchived)))
               "Scheduled refresh of archived model"))))))
+
+(deftest template-tag-referenced-card-ids-test
+  (testing "template-tag-referenced-card-ids returns the correct card ids from the query"
+    (mt/with-temp Card [card {:dataset_query {:database (mt/id)
+                                              :type     :native
+                                              :native   {:query         "select * from {{#14}} as a join {{#15}} as b on a.id = b.id"
+                                                         :template-tags {"#14" {:id           "some id"
+                                                                                :name         "some name"
+                                                                                :display-name "some name",
+                                                                                :type         "card"
+                                                                                :card-id      14}
+                                                                         "#15" {:id           "some id"
```

Comment:
nit: The id for each template-tags should be unique, even though we do not enforce it in the code but I think we should keep its uniqueness.

Here is an example of how we often name these params:
https://github.com/metabase/metabase/blob/988c0bfdc826cfebf4a1937cdffe5da6e5f12bd9/test/metabase/query_processor/card_test.clj#L68

Reply:
Okay, does this change fit what you expect?
https://github.com/metabase/metabase/pull/25197/commits/a1e2c5d4a1b57003fb3e4b8c9d2bd534be644d52